sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EnumerableSumInUnchecked.cs

```diff
@@ -48,7 +48,7 @@ protected override void Initialize(SonarAnalysisContext context)
                     var invocation = (InvocationExpressionSyntax)c.Node;
                     var expression = invocation.Expression;
                     if (expression is MemberAccessExpressionSyntax memberAccess &&
-                        invocation.ToStringContains("Sum") &&
+                        memberAccess.Name.Identifier.ValueText == "Sum" &&
```

Comment:
I would keep this , as for large methods, walking up the AST (IsSumInsideUnchecked) may be more intensive than doing a simple string comparison

however, it's a micro-optimization